todo_app/app.py

```diff
@@ -1,26 +1,69 @@
+import os
 from flask import Flask, render_template, request, redirect, url_for
+from flask_login import LoginManager, login_required, login_user
+import requests
 from todo_app.flask_config import Config
 from todo_app.data.mongodb_items import get_items, add_item, update_name, update_status, delete_item
+from todo_app.user import User
 from todo_app.view_model import ViewModel
 
 def create_app(environ=None, start_response=None):
     app = Flask(__name__)
     app.config.from_object(Config())
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        client_id=os.getenv('OAUTH_CLIENT_ID')
```

Comment:
Minor but it would be good to update your `.env.template` to make it easy to see what new variables need to be set!